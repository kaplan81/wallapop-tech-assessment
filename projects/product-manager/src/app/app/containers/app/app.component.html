<div class="root">
  <header>
    <mat-toolbar color="primary">
      <span>Lollipop</span>
      <button mat-raised-button>Favorites</button>
    </mat-toolbar>
  </header>
  <main><router-outlet></router-outlet></main>
</div>
