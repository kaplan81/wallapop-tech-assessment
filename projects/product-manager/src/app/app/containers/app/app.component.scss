.root {
  display: grid;
  grid-template-areas:
    'header'
    'main';
  grid-template-rows: 80px auto;
  grid-template-columns: 100%;
  width: 100%;
  height: 100%;
  header {
    grid-area: header;
    width: 100%;
    mat-toolbar {
      opacity: 75%;
      height: 100%;
      display: flex;
      justify-content: space-between;
      button {
        background-color: #fff;
      }
    }
  }
  main {
    grid-area: main;
  }
}
