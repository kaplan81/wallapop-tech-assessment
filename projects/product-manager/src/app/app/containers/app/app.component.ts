import { Component } from '@angular/core';

@Component({
  selector: 'mng-root',
  templateUrl: './app.component.html',
  standalone: true,
  styleUrls: ['./app.component.scss'],
})
export class AppComponent {
  title = 'product-manager';
}
