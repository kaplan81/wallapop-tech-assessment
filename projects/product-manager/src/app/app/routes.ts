import { Routes } from '@angular/router';
import { HomeComponent } from './containers/home/home.component';
import { ProductsComponent } from './containers/products/products.component';

export const routes: Routes = [
  {
    component: HomeComponent,
    path: '',
  },
  {
    component: ProductsComponent,
    path: 'products',
  },
];
