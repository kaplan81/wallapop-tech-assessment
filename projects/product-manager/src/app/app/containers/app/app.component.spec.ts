import { NgZone } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatDialog, MatDialogModule } from '@angular/material/dialog';
import { RouterTestingModule } from '@angular/router/testing';
import { Observable, of } from 'rxjs';
import { ShallowComponentSuite } from '../../models/component-suite.model';
import { ComponentSuite } from '../../test/component-suite';
import { ensureAngularZone } from '../../utils/ng.util';
import { AppComponent } from './app.component';

export class MatDialogMock {
  dialogResult?: any;
  returnedData: any = {};

  open(): {
    afterClosed: () => Observable<any>;
  } {
    console.log('OPENING');
    return {
      afterClosed: () => of(this.returnedData),
    };
  }
}
const matDialog = new MatDialogMock();

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let els: ShallowComponentSuite<AppComponent>;
  let dialog: MatDialog;
  let ngZone: NgZone;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent, RouterTestingModule, MatDialogModule],
      providers: [{ provide: MatDialog, useValue: matDialog }],
    }).compileComponents();
    fixture = TestBed.createComponent(AppComponent);
    els = new ComponentSuite(fixture).elements;
    dialog = TestBed.inject(MatDialog);
    els.host.component.dialog = dialog;
    ngZone = TestBed.inject(NgZone);
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should open dialog', () => {
    fixture.detectChanges();
    const openDialogSpy = jest.spyOn(matDialog, 'open');
    ensureAngularZone(ngZone, () => {
      els.host.component.openDialog();
    });
    expect(openDialogSpy).toHaveBeenCalled();
  });
});
