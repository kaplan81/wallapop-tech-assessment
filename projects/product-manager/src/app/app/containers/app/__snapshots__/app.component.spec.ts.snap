// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<mng-root
  __ngContext__="0"
  dialog={[Function MatDialogMock]}
  isFavouritesDisabled$={[Function Observable]}
  isRootRoute={[Function Boolean]}
>
  <div
    class="root"
  >
    <header>
      <mat-toolbar
        class="mat-toolbar mat-primary mat-toolbar-single-row"
        color="primary"
      >
        <span
          routerlink="/"
          tabindex="0"
        >
          ùë≥‚òâÔΩåÔΩåËÆ†—Ä‚òâ—Ä
        </span>
        <mng-search>
          <form
            class="search ng-untouched ng-pristine ng-valid"
            novalidate=""
          >
            <input
              class="search__input ng-untouched ng-pristine ng-valid"
              formcontrolname="searchCriteria"
              placeholder="Any search gets all products"
              type="text"
            />
          </form>
        </mng-search>
        <button
          class="mdc-button mdc-button--raised mat-mdc-raised-button mat-accent mat-mdc-button-base"
          color="accent"
          disabled="true"
          mat-raised-button=""
        >
          <span
            class="mat-mdc-button-persistent-ripple mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
             Favourites 
          </span>
          <span
            class="mat-mdc-focus-indicator"
          />
          <span
            class="mat-ripple mat-mdc-button-ripple"
            matripple=""
          />
          <span
            class="mat-mdc-button-touch-target"
          />
        </button>
      </mat-toolbar>
    </header>
    <main>
      <router-outlet />
    </main>
  </div>
</mng-root>
`;
